### ISSUE TITLE
feat: allow specifying peers to replicate from

### ISSUE THREAD

**Issue by iromli** [GluuFederation/docker-opendj#50]
The goals of this feature is to allow users to choose which peers are marked as replication source. These peers will be checked one by one.

There may be an issue with declaring such env because it may be misused by the user. We may attempt to do the following:
1. If the user declares a set of peers that do not exist the peer list will be ordered on a best effort basis based on existing peers.
1. If the user partially declares a set of peers that do exist , order on a best effort basis but giving the users input the priority. For example, if the user provided `ldap1`, `ldap0`, `ldap4` and only `ldap3`, `ldap4`, and `ldap5` actually existed the order should be `ldap4`, `ldap3` then `ldap5`.

For full description see https://github.com/GluuFederation/docker-opendj/issues/49#issuecomment-1260489387.

**Comment by iromli** [comment]
@moabu there is potential issue with this approach:


> If the user partially declares a set of peers that do exist , order on a best effort basis but giving the users input the priority.
> For example, if the user provided ldap1, ldap0, ldap4 and only ldap3, ldap4, and ldap5 actually existed the order should be ldap4, ldap3 then ldap5.

For illustration:

1. user provided `ldap1`, `ldap0`, `ldap4`
2. actual peers are `ldap3` (out-of-sync), `ldap4` (connection issue), `ldap5` (full entries)
3. `ldap1` and `ldap0` are not in actual peers so they are skipped
4. `ldap4` is having connection issue, so it's also skipped
5. `ldap3` has no connection issue, hence new container will replicate from out-of-sync container
6. `ldap5` is skipped because replication already established in prev step

The initial design is by having env `GLUU_LDAP_REPL_SOURCE=ldap1,ldap0,ldap4`, it implies that user really want to replicate from those peers (thus overrides the default behavior).

I may missed something but if you have different insight, please share it.

**Comment by iromli** [comment]
## Case 1

__Given:__ no user-defined peers (default strategy)
__Should:__ fallback to registered peers

```
/ # export GLUU_LDAP_REPL_PEERS=
/ # python3 /app/scripts/ldap_replicator.py 
INFO - ldap_replicator - 2022-09-29 20:03:03,958 - Getting current server info
INFO - ldap_replicator - 2022-09-29 20:03:04,193 - Force-enabling OpenDJ backend userRoot (if any).
INFO - ldap_replicator - 2022-09-29 20:03:06,490 - Force-enabling OpenDJ backend metric (if any).
INFO - ldap_replicator - 2022-09-29 20:03:08,887 - Force-enabling OpenDJ backend site (if any).
INFO - ldap_replicator - 2022-09-29 20:03:11,191 - Checking replicated backends (attempt 1)
INFO - ldap_replicator - 2022-09-29 20:03:11,191 - Detected user-defined replication peers=[]
INFO - ldap_replicator - 2022-09-29 20:03:11,195 - Registered replication peers=['ldap:7946', 'ldap2:7946']
INFO - ldap_replicator - 2022-09-29 20:03:11,195 - No user-defined replication peers; fallback to registered peers
INFO - ldap_replicator - 2022-09-29 20:03:11,195 - Replication peers=['ldap:7946', 'ldap2:7946']
```

## Case 2

__Given:__ no user-defined peers are matched
__Should:__ skip replication check thus allowing user to fix the ENV value later on

```
/ # export GLUU_LDAP_REPL_PEERS=ldap2--:7946,ldap-invalid
/ # python3 /app/scripts/ldap_replicator.py 
INFO - ldap_replicator - 2022-09-29 20:04:48,521 - Getting current server info
INFO - ldap_replicator - 2022-09-29 20:04:48,724 - Force-enabling OpenDJ backend userRoot (if any).
INFO - ldap_replicator - 2022-09-29 20:04:51,111 - Force-enabling OpenDJ backend metric (if any).
INFO - ldap_replicator - 2022-09-29 20:04:53,629 - Force-enabling OpenDJ backend site (if any).
INFO - ldap_replicator - 2022-09-29 20:04:56,240 - Checking replicated backends (attempt 1)
INFO - ldap_replicator - 2022-09-29 20:04:56,240 - Detected user-defined replication peers=['ldap2--:7946', 'ldap-invalid']
INFO - ldap_replicator - 2022-09-29 20:04:56,243 - Registered replication peers=['ldap:7946', 'ldap2:7946']
INFO - ldap_replicator - 2022-09-29 20:04:56,243 - Matched replication peers=[]
WARNING - ldap_replicator - 2022-09-29 20:04:56,244 - No matched replication peers; skipping replication check
```

## Case 3

__Given:__ 1 or more user-defined peers are matched
__Should:__ pick matched peers

```
/ # export GLUU_LDAP_REPL_PEERS=ldap2:7946,ldap-invalid
/ # python3 /app/scripts/ldap_replicator.py 
INFO - ldap_replicator - 2022-09-29 20:06:53,961 - Getting current server info
INFO - ldap_replicator - 2022-09-29 20:06:54,181 - Force-enabling OpenDJ backend userRoot (if any).
INFO - ldap_replicator - 2022-09-29 20:06:56,520 - Force-enabling OpenDJ backend metric (if any).
INFO - ldap_replicator - 2022-09-29 20:06:58,763 - Force-enabling OpenDJ backend site (if any).
INFO - ldap_replicator - 2022-09-29 20:07:01,056 - Checking replicated backends (attempt 1)
INFO - ldap_replicator - 2022-09-29 20:07:01,056 - Detected user-defined replication peers=['ldap2:7946', 'ldap-invalid']
INFO - ldap_replicator - 2022-09-29 20:07:01,060 - Registered replication peers=['ldap:7946', 'ldap2:7946']
INFO - ldap_replicator - 2022-09-29 20:07:01,060 - Matched replication peers=['ldap2:7946']
INFO - ldap_replicator - 2022-09-29 20:07:01,060 - Replication peers=['ldap2:7946']
```

**Comment by iromli** [comment]
Implemented in branch 4.2, 4.3, 4.4, and master
